### PR TITLE
Introducing cppclean scaning

### PR DESCRIPTION
**Pull request by atobiszei**
* Add script for veryfing count of warnings from cppclean
* Add additional step in sdl-check to check cppclean as well
* Remove unused forward declares, mark local functions as static
* Introduce forward declares instead of includes when possible

Script has hardcoded value of warnings. This is to be lowered with each
incoming fix. Several issues were fixed already.

There seems to be issue with false-positives when using c++ "using" directive in separate header files.
In this case we have to live with the issue reported by cppclean.

### CODE REVIEW

File: Dockerfile.redhat [hunk 1]
```diff
@@ -66,9 +66,9 @@ RUN dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.n
             which \
             yum-utils \
             unzip \
+            vim \
```
Comment (rasapala) [inline Dockerfile.redhat:69]: Is this needed ?

File: Dockerfile.ubuntu [hunk 2]
```diff
@@ -72,10 +72,10 @@ RUN apt update && apt install -y \
             unzip \
             wget \
             unzip \
+            vim \
```
Comment (rasapala) [inline Dockerfile.ubuntu:75]: Is this needed ?

File: src/custom_node_library_internal_manager_wrapper.hppmq [hunk 3]
```diff
@@ -0,0 +1,40 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
```
Comment (rasapala) [inline src/custom_node_library_internal_manager_wrapper.hppmq:2]: 2022

**Review by rasapala — APPROVED** [review]
LGTM minor comments.

**Review by michalkulakowski — APPROVED** [review]

File: Dockerfile.redhat (hunk 1 above)
Comment (atobiszei) [reply Dockerfile.redhat:69]: It's only for the build image. It's useful to have it there and not have to use apt to get it with setting proxy.

File: src/custom_node_library_internal_manager_wrapper.hppmq (hunk 3 above)
Comment (atobiszei) [reply src/custom_node_library_internal_manager_wrapper.hppmq]: File added by mistake. Removed.

**Pull request merged**